tests/nested/manual/core20-custom-kernel-commandline/task.yaml

```diff
@@ -28,26 +30,73 @@ prepare: |
     echo "Sign the shim binary"
     nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
 
-    echo 'hello from test' > pc-gadget/cmdline.extra
+    echo 'snapd.debug=1 hello from test' > pc-gadget/cmdline.extra
     snap pack pc-gadget/ extra-snaps/
 
+    echo 'snapd.debug=1 updated hello from test' > pc-gadget/cmdline.extra
+    snap pack pc-gadget --filename=pc-gadget-cmdline-extra-updated.snap
+
+    rm pc-gadget/cmdline.extra
+    echo 'snapd.debug=1 full hello from test' > pc-gadget/cmdline.full
+    snap pack pc-gadget --filename=pc-gadget-cmdline-full.snap
+
+    rm pc-gadget/cmdline.full
+    snap pack pc-gadget --filename=pc-gadget-cmdline-none.snap
+
     "$TESTSTOOLS"/nested-state build-image core
     "$TESTSTOOLS"/nested-state create-vm core
 
+    for f in pc-gadget-cmdline-extra-updated.snap pc-gadget-cmdline-full.snap pc-gadget-cmdline-none.snap; do
+        nested_copy "$f"
+    done
+
 execute: |
     #shellcheck source=tests/lib/nested.sh
     . "$TESTSLIB/nested.sh"
 
+    boot_id="$(nested_get_boot_id)"
+
+    echo "Make sure the system is encrypted"
+    # in which case the boot chains must exist
+    nested_exec sudo cat /var/lib/snapd/device/fde/boot-chains
+
     # system is in run mode
+    echo "Make sure the system is in run mode"
     nested_exec 'sudo cat /proc/cmdline' > system.cmdline.run
     MATCH snapd_recovery_mode=run < system.cmdline.run
 
     echo "Verify kernel command line in run mode"
     MATCH 'snapd_recovery_mode=run .* hello from test$' < system.cmdline.run
 
-    echo "Transition to recover mode to verify kernel command line"
+    echo "Perform an update of the gadget"
+    REMOTE_CHG_ID=$(nested_exec 'sudo snap install --dangerous --no-wait ./pc-gadget-cmdline-extra-updated.snap')
+
+    echo "Wait for reboot"
+    nested_wait_for_reboot "${boot_id}"
     boot_id="$(nested_get_boot_id)"
-    # shellcheck disable=SC2016
+    nested_exec 'sudo cat /proc/cmdline' | MATCH 'snapd_recovery_mode=run .* updated hello from test'
+    # wait for previous change to finish before proceeding
+    nested_exec sudo snap watch "$REMOTE_CHG_ID"
+
+    echo "Update to gadget with no command line"
+    REMOTE_CHG_ID=$(nested_exec 'sudo snap install --dangerous --no-wait ./pc-gadget-cmdline-none.snap')
+    nested_wait_for_reboot "${boot_id}"
+    boot_id="$(nested_get_boot_id)"
+    echo "Verify that custom command line elements are not present"
+    nested_exec 'sudo cat /proc/cmdline' | not MATCH 'hello from test'
+    # wait for previous change to finish before proceeding
+    nested_exec sudo snap watch "$REMOTE_CHG_ID"
+
+    # TODO: enable once full command line override is supported
```

Comment:
This will be enabled once https://github.com/snapcore/snapd/pull/10160 landed, yes?

Reply:
Yes